cluster/gce/addons/konnectivity-agent/konnectivity-agent-ds.yaml

```diff
@@ -33,6 +33,8 @@ spec:
                   "--ca-cert=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt",
                   "--proxy-server-host=__APISERVER_IP__",
                   "--proxy-server-port=8132",
+                  "--sync-interval=5s",
```

Comment:
Not sure 1s -> 5s will be enough. We can try and see how it goes.

Reply:
That's just reducing agent volume (for 5k cluster, change from 5k qps to 1k qps). The other file unsets proxy-server client-go throttling altogether (-1 means no qps limit). The auth requests seem very cheap and fast to handle, and this is local to the controlplane node.

Reply:
Agreed but 1k QPS is still a lot of extra QPS. Especially in the KAS was bounced case when we know the KAS can be load sensitive. At a minimum we may want to also increase the sync-interval-cap equivalently.

Reply:
Wait - are you saying that konnectivity agent can generate 1k QPS (i.e. request per sync-interval per agent)?

I don't think that's acceptable - that actually more-or-less doubles the number of requests. What are those requests? Why they need to be so frequent?

@mborsz @kubernetes/sig-scalability @jkaniuk

Reply:
To be clear we are talking about burst traffic and not sustained. The ANP agent tries to maintain a tunnel to each ANP Server. In a HA env I would expect there to be 3 ANP Servers. For this config we are using DS, so I would expect 1 agent per Node. When the ANP agent connects to a ANP server (for this config), the agent sends a token to authenticate itself. The ANP server will then validate that token with the KAS. Once validated the tunnel is established for that pair and a new request will not be needed unless the tunnel breaks. However when you have thousands of Nodes, this means thousands of agents and a lot of tunnels to establish. So we end up with an initial burst of validation requests from the ANP Server to the KAS.

Reply:
In GKE we switched away from DS to limit the max number of agents and for other reasons. We could always choose to do similar (Deployment with horizontal pod autoscaler) in this config.

Considerations with DS / large number of agents:
- Cluster bootstrap or scale-up can produce tall but short spikes, for which burst may help.
- During times when some ANP servers is down for maintenance/upgrade, there can be a shorter spike but lasting for minutes.
- We cannot have any one agent wait too long to sync, to be sure that small clusters can bootstrap quickly/reliably.

I opened https://github.com/kubernetes-sigs/apiserver-network-proxy/issues/237 to track ideas.

Reply:
> Once validated the tunnel is established for that pair and a new request will not be needed unless the tunnel breaks. However when you have thousands of Nodes, this means thousands of agents and a lot of tunnels to establish. So we end up with an initial burst of validation requests from the ANP Server to the KAS.

OK - thanks Walter. If that's only the initialization-related requests, then that sounds fine to me. Thanks for explanation.

Reply:
There is one slightly pathological case. In an HA cluster where 1 KAS (or more importantly an ANP Server) is down, the agents will periodically keep trying to connect to that server. There is a back-off to diminish the overall traffic level. It shouldn't be nearly as high as this limit but there be ongoing traffic in this case.

Reply:
OK - but that thing should be exercised also by rolling upgrade of the control-plane in HA scenario (which we test at least internally and are planning to also test in OSS).
Thanks for explanation!